test/codeu/chat/server/DerbyStoreTest.java

```diff
@@ -0,0 +1,190 @@
+package codeu.chat.server;
+
+import static org.junit.Assert.*;
+import org.junit.Test;
+import org.junit.Before;
+
+import codeu.chat.DerbyStore;
+import codeu.chat.common.User;
+import codeu.chat.util.Uuid;
+import codeu.chat.common.BasicController;
+import codeu.chat.common.Conversation;
+import codeu.chat.common.Message;
+
+public class DerbyStoreTest {
+	
+	private DerbyStore ds;
+	private Model model;
+	private BasicController controller;
+		
+	@Before
+	public void doBefore() {
+		ds = new DerbyStore();
+		model = new Model();
+	    controller = new Controller(Uuid.NULL, model);
+	}
+	
+	@Test
+	public void checkUserNameExists() {
+		final User user = controller.newUser("user");
+		boolean userNameExists = false;
+		
+		try {
+			userNameExists = ds.checkUsernameExists(user.name);
+		}
+		catch (Exception ex) {
+			
+		}
+		
+		assertTrue("Check to see if a username already exists",
+				userNameExists);
+	}
+	
+	@Test
+	public void addAndGetUser() {
+		 final User user = controller.newUser("user");
+		 User testUser = null;
+		 try {
+			//ds.addUser(user);
+			 testUser = ds.getUser(user.id);
```

Comment:
From a design perspective, it seems a little weird to go to the `DerbyStore` for some stuff and the `Controller` for other stuff. I would expect the `Controller` to sit on top of the `DerbyStore`, so you wouldn't go directly to the `DerbyStore` for anything?

Then I'd have two separate test classes: one for the `DerbyStore` that tests low-level saving and loading (and stuff like primary keys), and another one for `Controller` that uses a mocked `DerbyStore` to test the controller logic.